### PR TITLE
fix: router Executed event

### PR DESCRIPTION
**Pull request by dmoka**
## Checklist:

- [ ] I have updated the documentation if necessary.
- [ ] I have added tests to cover my changes, regression test if fixing an issue.
- [ ] This is a breaking change.

### CODE REVIEW

File: pallets/route-executor/src/lib.rs [hunk 1]
```diff
@@ -268,18 +270,17 @@ pub mod pallet {
...
 				T::Currency::reducible_balance(asset_in, &who, Preservation::Preserve, Fortitude::Polite);
 
 			let trade_amounts = Self::calculate_buy_trade_amounts(&route, amount_out)?;
 
-			let last_trade_amount = trade_amounts.last().ok_or(Error::<T>::RouteCalculationFailed)?;
-			ensure!(
-				last_trade_amount.amount_in <= max_amount_in,
-				Error::<T>::TradingLimitReached
-			);
+			let first_trade = trade_amounts.last().ok_or(Error::<T>::RouteCalculationFailed)?;
```
Comment (mrq1911) [inline pallets/route-executor/src/lib.rs:282]: last() is first? please add test to ensure you have this right

Comment (dmoka) [reply pallets/route-executor/src/lib.rs:282]: the `calculate_buy_trade_amounts` calculates the trades from the end as we have the `amount_out` specified. So the last of that list will be the `first_trade`, hence the renaming.

we cover this with test.

File: pallets/route-executor/Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = 'pallet-route-executor'
-version = '1.3.1'
+version = '1.3.2'
```
[the remark this answers is not in the document]
Comment (dmoka) [reply pallets/route-executor/Cargo.toml]: bumping it to 2 as it is breaking change.

File: pallets/route-executor/src/lib.rs (hunk 1 above)
Comment (mrq1911) [reply pallets/route-executor/src/lib.rs:282]: but funcionally you didnt change anything, just renamed `last_trade_amount` to `first_trade`

**Review by mrq1911 — APPROVED** [review]

File: pallets/route-executor/src/lib.rs (hunk 1 above)
Comment (mrq1911) [reply pallets/route-executor/src/lib.rs:282]: the problem is in the asset in vs route asset in inconsistency

**Pull request merged**